CUTLASS 3.0

jansel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94424

Not sure if we should land this, but curious what CI thinks of it.

jansel committed:
CUTLASS 3.0

Not sure if we should land this, but curious what CI thinks of it.

[ghstack-poisoned]

jansel commented on cmake/Dependencies.cmake:
```diff
@@ -1653,7 +1653,6 @@ if(NOT INTERN_BUILD_MOBILE)
   message(STATUS "Found CUDA with FP16 support, compiling with torch.cuda.HalfTensor")
   string(APPEND CMAKE_CUDA_FLAGS " -DCUDA_HAS_FP16=1"
                                  " -D__CUDA_NO_HALF_OPERATORS__"
-                                 " -D__CUDA_NO_HALF_CONVERSIONS__"
```
Newer versions of cutlass do not build cleanly with this flag.

Skylion007 replied:
@jansel Tried applying it to my PR. Let's see what happens.

Skylion007 replied:
@jansel Any idea how to fix this compile error? https://github.com/pytorch/pytorch/actions/runs/4127531837/jobs/7130854612

jansel replied:
Not sure.

jansel commented:
> @jansel CUTLASS 3.0 drops support for the Pascal architecture, which we still support. I made a PR to upgrade to 2.11 though here, but closed it because I didn't want to dig into the casting issues: #94188

The patches in this PR should fix the build issues in 2.11 too if you want to reopen your PR.  I am interested in performance testing the new features in 3.0, so 2.11 won't work for that.

Skylion007 commented:
> > @jansel CUTLASS 3.0 drops support for the Pascal architecture, which we still support. I made a PR to upgrade to 2.11 though here, but closed it because I didn't want to dig into the casting issues: #94188
> 
> The patches in this PR should fix the build issues in 2.11 too if you want to reopen your PR. I am interested in performance testing the new features in 3.0, so 2.11 won't work for that.

Already done.

jansel committed:
Update on "CUTLASS 3.0"


Not sure if we should land this, but curious what CI thinks of it.

[ghstack-poisoned]
